components/director/internal/dircleaner/service.go

```diff
@@ -0,0 +1,92 @@
+package dircleaner
+
+import (
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+	"github.com/kyma-incubator/compass/components/director/pkg/tenant"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// CleanerService missing godoc
+type CleanerService interface {
+	Clean(ctx context.Context) error
+}
+
+// TenantService missing godoc
+type TenantService interface {
+	ListSubaccounts(ctx context.Context) ([]*model.BusinessTenantMapping, error)
+	GetTenantByID(ctx context.Context, id string) (*model.BusinessTenantMapping, error)
+	Update(ctx context.Context, id string, tenantInput model.BusinessTenantMappingInput) error
+}
+
+// CisService missing godoc
+type CisService interface {
+	GetGlobalAccount(ctx context.Context, subaccountID string) (string, error)
+}
+
+type service struct {
+	transact  persistence.Transactioner
+	tenantSvc TenantService
+	cisSvc    CisService
+}
+
+// NewCleaner missing godoc
+func NewCleaner(transact persistence.Transactioner, tenantSvc TenantService, cisSvc CisService) CleanerService {
+	return &service{
+		transact:  transact,
+		tenantSvc: tenantSvc,
+		cisSvc:    cisSvc,
+	}
+}
+
+// Clean missing godoc
+func (s *service) Clean(ctx context.Context) error {
```

Comment:
It would be nice to add some logs in order to know what happens while the job is running as success is unrecoverable and hard to validate afterward (therefore you cannot do it again to see what happened).

Information like:
- How many subaccounts were listed
- Which subaccount is being processed
  - does it need an update or not
- For each updated - previous and new values

Then the full log of the first execution can be saved somewhere for reference (and potential recovery) in case something went wrong and we suspect that the job messed it up.

Also, double-check all of the errors if they should be fatal or not. I mean should we fail the whole job execution if one of the subaccounts enrichment fails? We should log it of course but I think we should proceed with next subaccount. Then we can aggregate the IDs of all the subaccounts that processing failed for them and print it at the end and we can proceed with either manual fixing them or execute the job again and validation if they are successfully fixed now.